src/settings/SettingTab.tsx

```diff
@@ -203,13 +211,71 @@ export class SmartCopilotSettingTab extends PluginSettingTab {
     const ollamaContainer = containerEl.createDiv(
       'smtcmp-settings-model-container',
     )
+    let modelDropdown: DropdownComponent | null = null // Store reference to the dropdown
+
+    const updateModelOptions = async (
+      baseUrl: string,
+      dropdown: DropdownComponent,
+    ) => {
```

Comment:
Please move this function to a private method of this class to improve readability.